R/data.table.R

```diff
@@ -1309,8 +1309,8 @@ replace_dot_alias = function(e) {
         jval = copy(jval)
       } else if ( length(jcpy <- which(vapply_1c(jval, address) %chin% vapply_1c(SDenv, address))) ) {
         for (jidx in jcpy) jval[[jidx]] = copy(jval[[jidx]])
-      } else if (jsub %iscall% 'get') {
-        jval = copy(jval) # fix for #1212
+      } else if (address(jval) %chin% vapply_1c(SDenv, address)) {
```

Comment:
sorry i don't follow how this fixes the issue, could you write a quick comment?

also if we're going to reuse the output of `vapply_1c(SDenv, address)` it might make sense to store that in a temporary variable for when we reach this branch

Reply:
When you call dt[, col] jval is just that column. It also gets assigned into SDenv. The first branch excludes list and the one before mine only checks the elements of a list (which would be columns in most other cases) but not the address of the list itself.

I thought about it but didn't think it was worth the hassle. I wouldn't want to have two assignments in a single if condition. Pushing those two branches below an else would impede clarity imo. But let me know if you disagree, I'm not married to that view.

Reply:
broke things into a more proper branch to make it more readable. i couldn't convince myself that this brunch is run only rarely, so trying to keep it streamlined in case.

Reply:
Thanks, looks good to me. I just replaced the assignment operators to keep with the style.